composer/trainer/trainer.py

```diff
@@ -108,6 +108,112 @@
 __all__ = ["Trainer"]
 
 
+class MissingArgumentException(ValueError):
+
+    def __init__(self, arg_name: str) -> None:
+        super().__init__((f"{arg_name} is a required argument and must be specified when constructing the "
+                          f"{Trainer.__name__} or when calling {Trainer.__name__}.{Trainer.fit.__name__}(). "
+                          f"To fix, please specify `arg_name` via {Trainer.__name__}({arg_name}=...) or "
+                          f"{Trainer.__name__}.{Trainer.fit.__name__}({arg_name}=...)."))
+
+
+def _scale_max_duration_by_ssr(scale_schedule_ratio: float,
+                               orig_max_duration: Optional[Time[int]]) -> Optional[Time[int]]:
+    if orig_max_duration is None:
+        return None
+    max_duration = cast(Time[int], orig_max_duration * scale_schedule_ratio)
+    log.info(f'max_duration changed from {orig_max_duration} to {max_duration}')
+    if max_duration.value == 0:
+        raise ValueError('Scale schedule has reduced the max_duration to 0. Set a higher ratio or use more epochs.')
+    return max_duration
+
+
+def _unpack_step_schedulers_every_batch(
+    schedulers: Union[None, ComposerScheduler, PyTorchScheduler, Sequence[Union[ComposerScheduler, PyTorchScheduler]]],
+    step_schedulers_every_batch: Optional[bool],
+):
```

Comment:
Seems like this function is missing a return type?

Reply:
PyRight can automatically infer return types, so they generally are not needed (except for abstract methods).